opentelekomcloud/resource_opentelekomcloud_obs_bucket_object.go

```diff
@@ -247,21 +271,27 @@ func resourceObsBucketObjectRead(d *schema.ResourceData, meta interface{}) error
 
 	class := string(object.StorageClass)
 	if class == "" {
-		d.Set("storage_class", "STANDARD")
+		err = d.Set("storage_class", "STANDARD")
 	} else {
-		d.Set("storage_class", normalizeStorageClass(class))
+		err = d.Set("storage_class", normalizeStorageClass(class))
+	}
+	mErr := multierror.Append(err,
+		d.Set("size", object.Size),
+		d.Set("etag", strings.Trim(object.ETag, `"`)),
+	)
+
+	if err := mErr.ErrorOrNil(); err != nil {
+		return fmt.Errorf("error setting OBS bucket attributes: %s", err)
 	}
-	d.Set("size", object.Size)
-	d.Set("etag", strings.Trim(object.ETag, `"`))
 
 	return nil
 }
 
 func resourceObsBucketObjectDelete(d *schema.ResourceData, meta interface{}) error {
 	config := meta.(*Config)
-	obsClient, err := config.newObjectStorageClient(GetRegion(d, config))
+	client, err := obsClient(d, config)
 	if err != nil {
-		return fmt.Errorf("Error creating OpenTelekomCloud OBS client: %s", err)
+		return err
```

Comment:
why you err without msg?

Reply:
Error message is specified in `obsClient` function: `errMsg := "error creating OpenTelekomCloud OBS client: %s"`